Suggested edits (#93)
LucidtechAI/cradl-docs: docs/concepts/training-data.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Training

A _training_ in Cradl is the process of training a machine learning model on the data you provide.
It can be summed up in 3 simple steps:
1. User requests training in the Cradl app, or with one of our SDKs or CLI.
2. Our team of Data Scientists verify that the training is feasible, and starts the training.
3. User receives a new model that has been trained according to the request.

:::info

> We make sure to always provide a model whose architecture is based on the latest research to make sure that our customers get the most out of their data.

Reviewer: Does it sound like we are not developing our own models?

Author: ```suggestion
We develop models whose architecture is based on the latest research to make sure that our customers get the most out of their data.
```

Reviewer: @staalezh Any comments?

Reviewer 2: Is this relevant for understanding the training process? Maybe we can elaborate on this elsewhere, e.g. in a separate technology section?

Author: A valid point, I was thinking a little promo wouldn't harm, but if no one else agree, I am open to omit it.

:::

## Request a training

The simplest way to request a training is to use the training wizard in the Cradl app, just go to [**Models**](https://app.cradl.ai/models) > **Your model** > **Training jobs** > **New training**.
If you want to request a training with one our SDKs or CLI you need to create a [**Data Bundle**](training-data.mdx#data-bundle) first.

<Tabs groupId="programming-language">
<TabItem value="cli" label="CLI" default>

```bash
las models create-training --name "Initial training of receipt model" <modelId> <dataBundleId>
```

</TabItem>
<TabItem value="curl" label="cURL">

```bash
curl -X POST 'https://api.lucidtech.ai/v1/models/<modelId>/trainings' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJra...' \
--data-raw '{
    "modelId": "<modelId>",
    "dataBundleIds": [
        "<dataBundleId>"
    ],
    "name": "Initial training of receipt model"
}'
```

</TabItem>
<TabItem value="python" label="Python">

```python
data_bundle = client.create_training(model_id='<modelId>', data_bundle_ids=['<dataBundleId>'], name='Initial training of receipt model')
```

</TabItem>
</Tabs>

> After the training is requested you will receive an email that verifies the request, and when the training is completed.

Reviewer: ```suggestion
You will be notified on email when a training is requested or changes status.
```

> You can also check the request of your training, by looking at its status.

Reviewer: ```suggestion
You can also check the status of your training by getting the training object with the CLI or in the Web app.
```

Author: ```suggestion
You can also check the status of your training by inspecting the training object with the CLI or in the Cradl app.
```

## Data Bundle

> A _data bundle_ in Cradl is a model-specific collection of one or more [Datasets](datasets.mdx).

Reviewer 2: ```suggestion
A _data bundle_ is a model-specific collection of one or more [Datasets](datasets.mdx).
```

Before training a model, you must specify a data bundle that the model will use as training data.

> Since the quality of the trained model depends on the quality of the data,
> we will make a [Data Report](training-data.mdx#data-report) that needs to be approved by one of our Data Scientists before the training can begin.

Reviewer 2: ```suggestion
Before the training job is started, one of our data scientists will review the auto-generated [Data Report](training-data.mdx#data-report) as part of our QA process.
```

> we require that any data bundles used for training must pass a statistical test in the form of a [Data Report](training-data.mdx#data-report) before training can begin.

Reviewer: What are the requirements for a statistical test to pass?

Author: They are often subject to change, so putting down something specific here would quickly be outdated I am afraid.

#### Good variation

> Machine learning algorithms are impressionable by nature, and can skew their predictions if the training data is skewed. One example of skewness is lack of variation: having too many similar examples in the training data. The variation of your data is quantified in the Variation statistic of the [Data bundle](training-data.mdx#data-report).

Author: ```suggestion
Machine learning algorithms are adaptable by design, and will learn patterns from the training data. One example of this is if the training data has a large number of similar training examples, the model may be encouraged to prioritize these documents at the expense of the less frequent ones. The variation of your data is quantified in the variation statistic of the [Data bundle](training-data.mdx#data-report).
```
